ddspipe_yaml/src/cpp/YamlReader_features.cpp

```diff
@@ -209,6 +211,102 @@ core::TopicRoutesConfiguration YamlReader::get(
     return object;
 }
 
+/*************************
+ * Monitor Configuration  *
+ **************************/
+
+template <>
+DDSPIPE_YAML_DllAPI
+void YamlReader::fill(
+        core::MonitorConfiguration& object,
+        const Yaml& yml,
+        const YamlReaderVersion version)
+{
+    // Optional domain
+    if (is_tag_present(yml, MONITOR_DOMAIN_TAG))
+    {
+        object.domain = get<int>(yml, MONITOR_DOMAIN_TAG, version);
+    }
+
+    /////
+    // Get optional monitor status tag
+    if (YamlReader::is_tag_present(yml, MONITOR_STATUS_TAG))
+    {
+        object.status = YamlReader::get<core::MonitorProducerConfiguration>(yml, MONITOR_STATUS_TAG, version);
+
+        if (!object.status.domain.is_set())
+        {
+            object.status.domain = object.domain;
+        }
+    }
+
+    /////
+    // Get optional monitor topics tag
+    if (YamlReader::is_tag_present(yml, MONITOR_TOPICS_TAG))
+    {
+        object.topics = YamlReader::get<core::MonitorProducerConfiguration>(yml, MONITOR_TOPICS_TAG, version);
+
+        if (!object.topics.domain.is_set())
+        {
+            object.topics.domain = object.domain;
+        }
+    }
+}
+
+template <>
+DDSPIPE_YAML_DllAPI
+core::MonitorConfiguration YamlReader::get(
+        const Yaml& yml,
+        const YamlReaderVersion version)
+{
+    core::MonitorConfiguration object;
+    fill<core::MonitorConfiguration>(object, yml, version);
+    return object;
+}
+
+template <>
+DDSPIPE_YAML_DllAPI
+void YamlReader::fill(
+        core::MonitorProducerConfiguration& object,
+        const Yaml& yml,
+        const YamlReaderVersion version)
+{
+    // Optional enable
+    if (is_tag_present(yml, MONITOR_ENABLE_TAG))
+    {
+        object.enabled = get<bool>(yml, MONITOR_ENABLE_TAG, version);
+    }
+
+    // Optional period
+    if (is_tag_present(yml, MONITOR_PERIOD_TAG))
+    {
+        object.period = get<uint32_t>(yml, MONITOR_PERIOD_TAG, version);
```

Comment:
Make sure this is greater than 0, we already have implemented methods checking this condition.